src/runtime_src/core/tools/common/XBHelpMenus.cpp

```diff
@@ -266,31 +266,34 @@ XBUtilities::produce_reports( xrt_core::device_collection devices,
         is_recovery = false;
       }
 
+      // Process the tests that require a device
+      // If the device is either of the following, most tests cannot be completed fully:
+      // 1. Is in factory mode and is not in recovery mode
+      // 2. Is not ready and is not in recovery mode
+      if((is_mfg || !is_ready) && !is_recovery) {
+        std::cerr << "Error: Device is not ready - Limited functionality available with XRT tools.\n\n";

```

Comment:
This shouldn't be an error, it should be a warning. It is completely ok to be able to run something like `xbmgmt examine -r platform` with golden or recovery image

Reply:
Makes sense to me. The return code is still non-zero due to the reports having issues while generating though